pwsh/Helpers.ps1

```diff
@@ -0,0 +1,138 @@
+#>
+function New-GitHubRelease {
+  param(
+    # An API token that provides authentication to GitHub
+    [string]$AuthToken,
+    # The SHA of the commit to create the tag and release on
+    [string]$GitSHA,
+    # The decscription of the release - usually release notes
+    [string]$ReleaseBody,
+    # The desired name for the release
+    [string]$ReleaseName,
+    # The name of the repository to create the release on
+    [string]$RepoName,
+    # The owner (username/organisation) of the repository to create the release on
+    [string]$RepoOwner,
+    # The name to give the created tag
+    [string]$TagName,
+    # Mark this release as a Draft release - do not publish it
+    [bool]$Draft,
+    # Mark this release as a pre-release
+    [bool]$PreRelease
+  )
+
+  $releases_url = "https://api.github.com/repos/$RepoOwner/$RepoName/releases"
+  $payload = @{
+    tag_name = "$TagName"
+    target_commitish = "$GitSHA"
+    name = "$ReleaseName"
+    body = "$ReleaseBody"
+    draft = $Draft
+    prerelease = $PreRelease
+  }
+
+  try {
+    [Net.ServicePointManager]::SecurityProtocol = "tls12, tls11, tls"
+    $result = `
+      Invoke-RestMethod `
+        -URI $releases_url `
+        -Headers @{Authorization = "token $AuthToken"} `
+        -Method 'POST' `
+        -ContentType "application/json" `
+        -Body ($payload | ConvertTo-Json)
+    return $result
+  } catch {
+    $_.Exception.Response
+    $_.ErrorDetails.Message
+    exit 1
+  }
+}
+
+<#
+.SYNOPSIS
+  This function is used to upload a file (asset) to a GitHub release.
+#>
+function Publish-ReleaseAsset() {
+  param(
+    # The name to give the file/asset on GitHub
+    [string]$AssetName,
+    # The path to the file to be uploaded
+    [string]$AssetPath,
+    # The type of file that is being uploaded (e.g. zip)
+    [string]$AssetType,
+    # An API token that provides authentication to GitHub
+    [string]$AuthToken,
+    # The ID number of the release to publish the asset to
+    [string]$ReleaseID,
+    # The name of the repository to create the release on
+    [string]$RepoName,
+    # The owner (username/organisation) of the repository to create the release on
+    [string]$RepoOwner
+  )
+
+  $upload_url = "https://uploads.github.com/repos/{0}/{1}/releases/{2}/assets"
+  $upload_url = $upload_url -f @("$RepoOwner", "$RepoName", "$ReleaseID")
+  $full_upload_url = "${upload_url}?name=${AssetName}"
+
+  [Net.ServicePointManager]::SecurityProtocol = "tls12, tls11, tls"
+  try {
+    $result = Invoke-RestMethod `
+                -URI "$full_upload_url" `
+                -Headers @{Authorization = "token $AuthToken"} `
+                -Method 'POST' `
+                -ContentType "application/$AssetType" `
+                -InFile "$AssetPath"
+  } catch {
+    $_.Exception
+    $_.ErrorDetails.Message
+    exit 1
+  }
+  return $result
+}
+
+<#
+.SYNOPSIS
+  Function to deduce the content type of the given file. Only .zip and .tar.gz
```

Comment:
You say "content type" in the comment, "applicationType" in the function name and "assetType" in the variable set to the return value. What we're defining is the content-type  of the asset which is a MIME type.... The language is a little confused.

The comment says `.tar.gz` but the code below is just checking `.gz`. As the MIME Type only recognises `tar`, `zip` or `gz` I think it's the comment that needs relaxing